src/Domain/CoordinateMaps/EquatorialCompression.cpp

```diff
@@ -54,42 +62,49 @@ EquatorialCompression::angular_distortion(const std::array<T, 3>& coords,
     // a value of zero) in the case rho==0.
   }
-  return std::array<ReturnType, 3>{{radius_over_rho * x, radius_over_rho * y,
-                                    inverse_alpha * radius_over_rho * z}};
+  return std::array<ReturnType, 3>{
+      {radius_over_rho * ((index_pole_axis == 0) ? inverse_alpha * x : x),
+       radius_over_rho * ((index_pole_axis == 1) ? inverse_alpha * y : y),
+       radius_over_rho * ((index_pole_axis == 2) ? inverse_alpha * z : z)}};
 }
 
 template <typename T>
 tnsr::Ij<tt::remove_cvref_wrap_t<T>, 3, Frame::NoFrame>
 EquatorialCompression::angular_distortion_jacobian(
-    const std::array<T, 3>& coords, const double inverse_alpha) const noexcept {
+    const std::array<T, 3>& coords, const double inverse_alpha,
+    const size_t index_pole_axis) const noexcept {
   using ReturnType = tt::remove_cvref_wrap_t<T>;
   const ReturnType& x = coords[0];
   const ReturnType& y = coords[1];
   const ReturnType& z = coords[2];
   const ReturnType radius = sqrt(square(x) + square(y) + square(z));
   const ReturnType rho =
-      sqrt(square(x) + square(y) + square(inverse_alpha * z));
+      sqrt(square((index_pole_axis == 0) ? inverse_alpha * x : x) +
+           square((index_pole_axis == 1) ? inverse_alpha * y : y) +
+           square((index_pole_axis == 2) ? inverse_alpha * z : z));
+
   // Various common factors:
-  const ReturnType lambda1 = (pow<2>(inverse_alpha) - 1.0) * z * z;
-  const ReturnType lambda2 = (1.0 - pow<2>(inverse_alpha)) * (x * x + y * y);
+  const ReturnType lambda1 =
+      (pow<2>(inverse_alpha) - 1.0) * square(coords[index_pole_axis]);
+  const ReturnType lambda2 = (1.0 - pow<2>(inverse_alpha)) *
+                             (square(radius) - square(coords[index_pole_axis]));
   const ReturnType lambda3 = pow<2>(rho * radius);
   const ReturnType lambda4 = radius * pow<3>(rho);
 
   auto jacobian_matrix =
       make_with_value<tnsr::Ij<tt::remove_cvref_wrap_t<T>, 3, Frame::NoFrame>>(
           dereference_wrapper(coords[0]), 0.0);
 
-  get<0, 0>(jacobian_matrix) = (x * x * lambda1 + lambda3);
-  get<1, 0>(jacobian_matrix) = x * y * lambda1;
-  get<2, 0>(jacobian_matrix) = inverse_alpha * x * z * lambda1;
-
-  get<0, 1>(jacobian_matrix) = get<1, 0>(jacobian_matrix);
-  get<1, 1>(jacobian_matrix) = (y * y * lambda1 + lambda3);
-  get<2, 1>(jacobian_matrix) = inverse_alpha * y * z * lambda1;
-
-  get<0, 2>(jacobian_matrix) = x * z * lambda2;
-  get<1, 2>(jacobian_matrix) = y * z * lambda2;
-  get<2, 2>(jacobian_matrix) = inverse_alpha * (z * z * lambda2 + lambda3);
+  const ReturnType zero = make_with_value<ReturnType>(x, 0.0);
+  for (size_t i = 0; i < 3; i++) {
+    for (size_t j = 0; j < 3; j++) {
+      const size_t k = (index_pole_axis + j) % 3;
+      jacobian_matrix.get(i, k) =
+          (i == index_pole_axis ? inverse_alpha : 1.0) *
+          (coords[i] * coords[k] * (j == 0 ? lambda2 : lambda1) +
+           (i == k ? lambda3 : make_with_value<ReturnType>(x, 0.0)));
```

Comment:
If I understand this correctly, then I think it's better to split this addition into a separate statement under an `if (i==k)` clause, so you don't need the zero vector.

Reply:
This addition is inside the multiplication by `(i == index_pole_axis ? inverse_alpha : 1.0)` so I will also need to factor that out of this loop as well, so it would look like:

```
jacobian_matrix.get(i, k) =
          coords[i] * coords[k] * (j == 0 ? lambda2 : lambda1);
      if(i== k){
        jacobian_matrix.get(i, k) += lambda3;
      }
      if(i == index_pole_axis){
        jacobian_matrix.get(i, k) *= inverse_alpha;
      }
```

I can change it to this, I agree it does look clearer